overlord/snapstate/check_snap.go

```diff
@@ -180,13 +184,165 @@ func validateInfoAndFlags(info *snap.Info, snapst *SnapState, flags Flags) error
 	return nil
 }
 
+func normPath(path string) string {
+	if path == "" {
+		return ""
+	}
+
+	path = strings.TrimPrefix(filepath.Clean(path), "/")
+	if strings.HasPrefix(path, "../") {
```

Comment:
Hm, some questions: when does this condition happens? And if we don't handle it, is `normPath()` as a name maybe a bit too generic? I.e. if I call `normPath("../foo")` the result `""` is a bit unexpected.

Reply:
you can create a snap with an app whose command is `../../../bin/sh`, in which case there's nothing for the checker to check.
`normPath` signals "nothing to do" with the empty string.
maybe i should add a comment :-)

Reply:
It would be nice if we had a bit of logic that reconstructs the real path knowing two bits of information:

1) classic flag for the snap
2) effective mount dir for the snap (which depends on 1)

Then we can construct an absolute path by normalizing whatever relative paths we get here and complete the validation.